### PR TITLE
Add `id`, `created_at` and `updated_at` to data models

### PR DESCRIPTION
**Pull request by The0mikkel**
This update updates the data models to now include `id`, `created_at` and `updated_at`.

This therefore solves #33.

Please note, that this is not a breaking change, as it is additive.

### CODE REVIEW

File: src/utils/Api/fetchCars.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        setState({
+            cars: [],
+            error: `Error: Invalid content type: ${contentType}`,
+            loading: false,
+        });
+        return;
+    }
+
+    // Parse response body as json
+    const cars = await response.json() as unknown;
```
Comment (OthelloEngineer) [inline src/utils/Api/fetchCars.ts:36]: Seems hella sus with unknown for type safety concerns, but seems like there are no fixes...
https://github.com/node-fetch/node-fetch/issues/1262

**Review by OthelloEngineer — COMMENTED** [review]
Also: 
Shouldn't ID's in zod be string().uuid() or number().nonnegative()?

Either we use UUID or ULID which are subtypes from strings or we should use numeric/serial ID's whicv are nonnegative.

File: src/utils/Api/fetchCars.ts (hunk 1 above)
Comment (The0mikkel) [reply src/utils/Api/fetchCars.ts:36]: This have been discussed, and is the correct implementation.

**Comment by The0mikkel** [comment]
> Also: Shouldn't ID's in zod be string().uuid() or number().nonnegative()?
> 
> Either we use UUID or ULID which are subtypes from strings or we should use numeric/serial ID's whicv are nonnegative.

We could indeed go with a UUID or other "string" based id format.  
I think we would have to discuss this further in the group, but I don't feel that for this kind of exercise ***needs*** uuids, and simple integer based ids is good enough.

**Review by JfriisJ — APPROVED** [review]
its fine, but context is missing description

**Pull request closed without merging**